pyxem/signals/diffraction2d.py

```diff
@@ -466,12 +473,23 @@ def get_direct_beam_position(self, method, **kwargs):
 
         method_function = select_method_from_method_dict(method, method_dict, **kwargs)
 
+        if square_width is not None:
+            min_index = np.int(origin_coordinates[0] - square_width)
+            # fails if non-square dp
+            max_index = np.int(origin_coordinates[0] + square_width)
+            sig = self.isig[min_index:max_index, min_index:max_index]
+        else:
```

Comment:
Adding this functionality here might require us to change how center_direct_beam() is written....